src/interfaces/IRouter.sol

```diff
@@ -1,26 +1,29 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import {BLS} from "../libraries/BLS.sol";
+
 interface IRouter {
     // -------- Structs --------
 
-    struct TransferParams {
+    struct SwapRequestParameters {
         address sender; // Address initiating the swap on the source chain
         address recipient; // Address to receive tokens on the destination chain
         address token; // Token address being transferred
         uint256 amount; // Amount to be received by the recipient on the destination chain
         uint256 srcChainId; // Source chain ID where the request originated
         uint256 dstChainId; // Destination chain ID where tokens will be delivered
-        uint256 swapFee; // Total swap fee deducted from the amount
-        uint256 solverFee; // Portion of swapFee paid to the solver
+        uint256 verificationFee; // Total swap fee deducted from the amount
```

Comment:
I thought this was just the fee paid to dcipher, and the total fee was $verificationFee + solverFee$.

Reply:
That's right. The total fee is verification fee + solver fee.

We can pass in the solver fee on it's own and deduct the verification fee based on the amount being swapped @CluEleSsUK